File: 백준/Gold/1197. 최소 스패닝 트리/최소 스패닝 트리.py
import sys

V,E=map(int, sys.stdin.readline().split())
graph=[[]for _ in range(E)]
parent=[i for i in range(V+1)]
mst=0

for i in range(E):
    a,b,c=map(int, sys.stdin.readline().split())
    graph[i]=[a,b,c]

graph.sort(key=lambda x : x[2])

def find(x):
    if x==parent[x]:
        return parent[x]
    return find(parent[x])
for g in graph:
    x=g[0]
    y=g[1]
    px=find(x)
    py=find(y)
    if px!=py:
        if px<py:
            parent[py]=px

        elif px>py:
            parent[px]=py

        mst+=g[2]
    if parent[1:]==[1]:
        break

print(mst)

File: 백준/Gold/1197. 최소 스패닝 트리/최소 스패닝 트리.java
import java.io.*;
import java.util.*;


public class Main {

    public static int find(int x,int[] parent){
        if (parent[x]==x)
            return parent[x];
        return find(parent[x],parent);
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br= new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st= new StringTokenizer(br.readLine());
        int V=Integer.parseInt(st.nextToken());
        int E=Integer.parseInt(st.nextToken());

        int[][] graph= new int[E][3];
        int[] parent = new int[V+1];

        for(int i=0;i<E;i++){
            st=new StringTokenizer(br.readLine());
            graph[i][0]=Integer.parseInt(st.nextToken());
            graph[i][1]=Integer.parseInt(st.nextToken());
            graph[i][2]=Integer.parseInt(st.nextToken());
        }
        for(int i=0;i<V+1;i++){
            parent[i]=i;
        }

        Arrays.sort(graph, (o1,o2)->(o1[2]-o2[2]));

        int cnt=0;
        for(int i=0;i<E;i++){
            int p0=find(graph[i][0],parent);
            int p1=find(graph[i][1],parent);
            if (p0!=p1){
                if (p0>p1){
                    parent[p0]=p1;
                }
                else if (p0<p1){
                        parent[p1]=p0;
                }
                cnt+=graph[i][2];
            }
        }
        System.out.println(cnt);
    }

}